web/frontend/src/pages/election/Index.tsx

```diff
@@ -17,30 +29,151 @@ const ElectionIndex: FC = () => {
 
-  /*Show all the elections retrieved if any */
-  const showElection = () => {
-    return data.Elections.length > 0 ? (
-      <>
-        <div className="pb-8">
-          <h2 className="text-2xl font-bold leading-7 text-gray-900 sm:text-3xl sm:truncate">
+  // Apply the filter statusToKeep
+  useEffect(() => {
+    if (data !== null) {
+      if (statusToKeep !== null) {
+        const filteredElectionsID = [];
+        electionStatuses.forEach((status, id) => {
+          if (status === statusToKeep) {
+            filteredElectionsID.push(id);
+          }
+        });
+
+        const filteredElections = (data.Elections as LightElectionInfo[]).filter((election) =>
+          filteredElectionsID.includes(election.ElectionID)
+        );
+        setElections(filteredElections);
+      } else {
+        setElections(data.Elections);
+      }
+    }
+  }, [data, statusToKeep]);
+
+  // Fetch the NodeStatus for each node of each election
+  useEffect(() => {
+    if (elections !== null) {
+      const fetchDKGStatus = async (id: ID, node: string, proxyAddress: string) => {
+        try {
+          const response = await fetch(endpoints.getDKGActors(proxyAddress, id), request);
+          if (!response.ok) {
+            // The node is not initialized
+            if (response.status === 404) {
+              return { ID: id, Node: node, Status: NodeStatus.NotInitialized };
+            } else {
+              const js = await response.json();
+              fctx.addMessage(JSON.stringify(js), FlashLevel.Error);
+            }
+          } else {
+            let dkgStatus = await response.json();
+
+            return { ID: id, Node: node, Status: dkgStatus.Status as NodeStatus };
+          }
+        } catch (e) {
+          fctx.addMessage(e.message, FlashLevel.Error);
+        }
+      };
+
+      const fetchProxies = async (election: LightElectionInfo) => {
+        try {
+          const response = await fetch(endpoints.getProxiesAddresses(election.ElectionID), request);
+          if (!response.ok) {
+            const js = await response.json();
+            throw new Error(JSON.stringify(js));
+          } else {
+            let dataReceived = await response.json();
+
+            const newNodeProxyAddresses = new Map<string, string>();
+            dataReceived.Proxies.forEach((value) => {
+              Object.entries(value).forEach(([node, proxy]) => {
+                newNodeProxyAddresses.set(node, proxy as string);
+              });
+            });
+
+            return { ID: election.ElectionID, NodeProxy: newNodeProxyAddresses };
+          }
+        } catch (e) {
+          fctx.addMessage(e.message, FlashLevel.Error);
+        }
+      };
+
+      const promises = elections
+        .filter((election) => election.Status === Status.Initial)
+        .map((election) => {
+          return fetchProxies(election);
+        });
+
+      Promise.all(promises)
```

Comment:
Yes this is indeed problematic. I think I tried to synchronize them because there was some weird things happening with the states (the status was not re-rendering on each update, etc...). I have just refactored the code so there's no synchronization anymore. But I don't know when to set the DKGLoading to False (e.g all the elections that needs to have the status of their nodes have them). So the status sometimes changes after the page has loaded...